python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -365,6 +373,33 @@ def run(self, science, matchedTemplate, difference,
                                    positiveFootprints=positives,
                                    negativeFootprints=negatives)
 
+    def _prepareInputs(self, difference):
+        """Ensure that we start with an empty detection and deblended mask.
+
+        Parameters
+        ----------
+        difference : `lsst.afw.image.ExposureF`
+            The difference image that will be used for detecting diaSources.
+            The mask plane will be modified in place.
+
+        Returns
+        -------
+        sigma : `float`
+            The width of the PSF of the difference image.
+        """
+        # Ensure that we start with an empty detection and deblended mask.
+        mask = difference.mask
+        for mp in self.config.clearMaskPlanes:
+            if mp not in mask.getMaskPlaneDict():
+                mask.addMaskPlane(mp)
+        mask &= ~mask.getPlaneBitMask(self.config.clearMaskPlanes)
+        sigma = difference.psf.computeShape(difference.psf.getAveragePosition()).getDeterminantRadius()
+        if np.isnan(sigma):
+            sigma = self.config.fallbackPsfSigma
+            self.log.warning("Could not determine average width of the difference image PSF. "
+                             "Using the default sigma of %d", self.config.fallbackPsfSigma)
```

Comment:
Why can't we make this be an `UpstreamFailureNoWorkFound`, and do it as the first step of this method? See DM-46948 for details, but this case is exactly what that exception was made for.

Not having a useable PSF on the diffim seems like a totally reasonable failure condition. I don't like using some default psf size, since that will result in unreliable measurements and we're not doing anything to mark the output catalog as being untrustworthy.

Reply:
I went back and forth on this, so I am fine checking for a valid PSF here and raising an error. I will look into `UpstreamFailureNoWorkFound`